virtualpdu/device_provider.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2016 Internap
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class DeviceProvider(object):
+    def power_on(self, name):
+        raise NotImplementedError()
+
+    def power_off(self, name):
+        raise NotImplementedError()
+
```

Comment:
I don't want to give them the choice, only to see every implementation do "power_off" "power_on". I will enforce this in the core.